src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -431,28 +432,142 @@ private static bool OverloadResolutionResultIsValid<TMember>(ArrayBuilder<Member
+
+                        switch (unmanagedCallingConventionTypes.Count)
+                        {
+                            case 0:
+                                actualCallKind = Cci.CallingConvention.Unmanaged;
+                                actualUnmanagedCallingConvetionTypes = ImmutableHashSet<INamedTypeSymbolInternal>.Empty;
+                                break;
+                            case 1:
+                                switch (unmanagedCallingConventionTypes.Single().Name)
+                                {
+                                    case "CallConvCdecl":
+                                        actualCallKind = Cci.CallingConvention.CDecl;
+                                        actualUnmanagedCallingConvetionTypes = ImmutableHashSet<INamedTypeSymbolInternal>.Empty;
+                                        break;
+                                    case "CallConvStdcall":
+                                        actualCallKind = Cci.CallingConvention.Standard;
+                                        actualUnmanagedCallingConvetionTypes = ImmutableHashSet<INamedTypeSymbolInternal>.Empty;
+                                        break;
+                                    case "CallConvThiscall":
+                                        actualCallKind = Cci.CallingConvention.ThisCall;
+                                        actualUnmanagedCallingConvetionTypes = ImmutableHashSet<INamedTypeSymbolInternal>.Empty;
+                                        break;
+                                    case "CallConvFastcall":
+                                        actualCallKind = Cci.CallingConvention.FastCall;
+                                        actualUnmanagedCallingConvetionTypes = ImmutableHashSet<INamedTypeSymbolInternal>.Empty;
+                                        break;
+                                    default:
+                                        goto outerDefault;
+                                }
+                                break;
+
+                            default:
+outerDefault:
+                                actualCallKind = Cci.CallingConvention.Unmanaged;
+                                actualUnmanagedCallingConvetionTypes = unmanagedCallingConventionTypes;
+                                break;
+                        }
+                    }
+
+                    // The rules for matching a calling convention are:
+                    // 1. The CallKinds must match exactly
+                    // 2. If the CallKind is Unmanaged, then the set of calling convention types must match exactly, ignoring order
+                    //    and duplicates. We already have both sets in a HashSet, so we can just ensure they're the same length and
+                    //    that everything from one set is in the other set.
+
+                    if (!actualCallKind.IsCallingConvention(expectedConvention.CallKind))
```

Comment:
>!actualCallKind.IsCallingConvention(expectedConvention.CallKind) [](start = 24, length = 64)

It looks like ConversionBase.HasImplicitPointerConversion simply uses equality check. Is there a reason to differ?

Reply:
Yeah, we could change this. I'll add an assert that the expected convention has no unknown bits in it.

---
In reply to: [491116706](https://github.com/dotnet/roslyn/pull/47651#discussion_r491116706) [](ancestors = 491116706)

Reply:
@AlekseyTs update: `member.CallingConvention` here can include both a CallKind and a possible calling convention attribute (such as Generic/HasThis/ExplicitThis). HasImplicitPointerConversion doesn't need to worry about these, because those are both function pointer types and won't have a convention attribute. However, we do have to worry about them in this case because a regular method may have them, and `IsCallingConvention` will do the correct comparison here.

---
In reply to: [491132506](https://github.com/dotnet/roslyn/pull/47651#discussion_r491132506) [](ancestors = 491132506,491116706)